packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -48,6 +51,155 @@ export default class MuiDetailGenerator implements DetailGenerator {
     );
   }
 
+  async getFormWidgetProperties(position: SourceLineCol): Promise<WidgetProperties> {
+    let result: WidgetProperties = {
+      properties: []
+    };
+
+    if (this._widgetContext) {
+      const sourceCode = await this._widgetContext.getSourceCodeString(position);
+      const ast = createAst(sourceCode);
+
+      if (ast) {
+        result = this.getFormWidgetPropertiesFromAst(ast, position);
+      }
+    }
+
+    return result;
+  }
+
+  getFormWidgetPropertiesFromAst(ast: ts.SourceFile, position: SourceLineCol): WidgetProperties {
+    const result: WidgetProperties = {
+      properties: []
+    };
+
+    const pos = ast.getPositionOfLineAndCharacter(position.lineNumber - 1, position.columnNumber - 1);
+    const element = findByCondition<Node>(ast, (node: Node) => {
+      return node.pos === pos;
+    });
+
+    if (element) {
+      if (ts.isJsxOpeningElement(element) || ts.isJsxSelfClosingElement(element)) {
+        element.attributes.properties.forEach(prop => {
+          if (ts.isJsxAttribute(prop)) {
+            const propName = prop.name.escapedText.toString();
+
+            if (prop.initializer) {
+              if (ts.isStringLiteral(prop.initializer)) {
+                result.properties.push({
+                  name: propName,
+                  value: prop.initializer.text
+                });
+              }
+              else if (ts.isJsxExpression(prop.initializer)) {
+                if (prop.initializer.expression) {
+                  if (ts.isNumericLiteral(prop.initializer.expression)
+                    || prop.initializer.expression.kind === SyntaxKind.TrueKeyword
+                    || prop.initializer.expression.kind === SyntaxKind.FalseKeyword) {
+                    result.properties.push({
```

Comment:
please use `array = [...array, newItem]` instead of `push()`

Reply:
fixed..btw is is only about style of coding or does it have some other advanteges?

Reply:
it is just preferred in the javascript community